Automattic/jetpack
# Jetpack Social: Add TemplatePicker component

## Proposed changes:
This PR adds a new `TemplatePicker` component, that can be used to choose a new template for a generated image. The component is functionally complete, but isn't properly implemented anywhere yet. I added it to the SIG sidebar for testing purposes but it's not actually functional there yet.

### Other information:


## Jetpack product discussion


1202583289493814-as-1204097909752716/f

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






* Create a new post and open the Social sidebar.
* Scroll down to the SIG settings and press the "Change Template" button.
* Make sure the template picker matches the screenshot.
* Make the window smaller and check that it is responsive.
* Run the tests: `cd projects/js-packages/publicize-components && npm test src/components/social-image-generator/template-picker/test/index.test.js`

<img width="1089" alt="image" src="https://user-images.githubusercontent.com/1713699/225687996-9f6bce9c-f13c-4cd4-a7e2-b7f3654f9738.png">
